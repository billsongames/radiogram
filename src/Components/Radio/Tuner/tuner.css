.tuner__container {
  width: 1200px;
  height: 360px;
  margin: auto;
  background-image: url("../../../assets/img/bg_tuner-surround_2.png");
  border-radius: 8px;
}

.tuner__main {
  width: 1200px;
  margin: auto;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.tuner__title {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: auto;
  text-align: left;
  font-size: 18px;
  font-weight: bold;
  padding: 8px;
  color: whitesmoke;
}

.tuner-carousel {
  margin: auto;
  width: 1200px;
  height: 160px;
  padding: 24px 0px 0px
  
}

.carousel-entry {
  width: 140px;
  height: 140px;
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  z-index: 32;

}  

/* .carousel-entry:hover{
  background-color: grey;
  transition-duration: 0.5s;
} */

.tuner-station__logo {
  width: 80px;
  height: 80px;
  object-fit: contain;
  border-radius: 8px;
  cursor: pointer;
}

.tuner-station__name {
  font-size: 14px;
  color: whitesmoke;
}

.tuner-filter__container {
  margin: auto;
  padding: 4px 0px 4px 0px;
  display: flex;
  flex-direction: row;
  align-items: center;
}

.tuner-filter__item {
  font-size: 14px;
  font-weight: bold;
  color: whitesmoke;
  padding: 4px 10px;
  cursor: pointer
}


.filter__button {
  -webkit-appearance: none;
  appearance: none;
  position: relative;
  border-width: 0;
  padding: 0 4px 4px;
  width: 120px;
  box-sizing: border-box;
  background: transparent;
  font: inherit;
  cursor: pointer;
}

.filter__button-top {
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
  z-index: 0;
  padding: 8px 16px;
  transform: translateY(0);
  text-align: center;
  color: whitesmoke;
  text-shadow: 0 -1px rgba(0, 0, 0, .25);
  transition-property: transform;
  transition-duration: .1s;
  -webkit-user-select: none;
  user-select: none;
}

.filter__button:active .filter__button-top {
  transform: translateY(4px);
}

.filter__button-top::after {
  content: '';
  position: absolute;
  z-index: -1;
  border-radius: 4px;
  width: 100%;
  height: 80%;
  box-sizing: content-box;
  background-color: black;
  text-align: center;
  color: #fff;
  box-shadow: inset 0 0 0px 1px rgba(255, 255, 255, .2), 0 1px 2px 1px rgba(255, 255, 255, .2);
  transition-property: border-radius, padding, width, transform;
  transition-duration: .1s;
}

.filter__button:active .filter__button-top::after {
  border-radius: 6px;
  padding: 0 2px;
}

.filter__button-bottom {
  position: absolute;
  z-index: -1;
  bottom: 4px;
  left: 4px;
  border-radius: 8px / 16px 16px 8px 8px;
  padding-top: 6px;
  width: calc(100% - 8px);
  height: calc(100% - 10px);
  box-sizing: content-box;
  background-color: grey;
  background-image: radial-gradient(4px 8px at 4px calc(100% - 8px), rgba(255, 255, 255, .25), transparent), radial-gradient(4px 8px at calc(100% - 4px) calc(100% - 8px), rgba(255, 255, 255, .25), transparent), radial-gradient(16px at -4px 0, white, transparent), radial-gradient(16px at calc(100% + 4px) 0, white, transparent);
  box-shadow: 0px 2px 3px 0px rgba(0, 0, 0, 0.5), inset 0 -1px 3px 3px rgba(0, 0, 0, .4);
  transition-property: border-radius, padding-top;
  transition-duration: .2s;
}

.filter__button:active .filter__button-bottom {
  border-radius: 10px 10px 8px 8px / 8px;
  padding-top: 0;
}

.filter__button-base {
  position: absolute;
  z-index: -2;
  top: 4px;
  left: 0;
  border-radius: 12px;
  width: 100%;
  height: calc(100% - 4px);
  background-color: rgba(0, 0, 0, .15);
  box-shadow: 0 1px 1px 0 rgba(255, 255, 255, .75), inset 0 2px 2px rgba(0, 0, 0, .25);
}

.tuner__ButtonGroup {
  padding: 32px 0px;
  position: absolute;
  width: 1200px;
  display: flex;
  justify-content: center
}

.tuner__button {
  padding: 0px 8px;
  cursor: pointer
}
