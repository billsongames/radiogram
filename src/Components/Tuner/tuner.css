.tuner {
  margin: 25%;
  width: 50%;
}

