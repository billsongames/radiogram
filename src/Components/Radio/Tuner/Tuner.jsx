import { React, useCallback, useEffect, useState } from 'react';
import { RadioBrowserApi } from "radio-browser-api";

import { IconContext } from "react-icons";
import { BsArrowLeftSquareFill, BsArrowRightSquareFill } from "react-icons/bs";


import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";

import { radio_search_filters } from '../../../data/filters';

import default_station_logo from "../../../assets/img/station_no_logo_2.png"

import "./tuner.css"

const responsive = {
  superLargeDesktop: {
    // the naming can be any, depends on you.
    breakpoint: { max: 4000, min: 1201 },
    items: 8,
    partialVisibilityGutter: 0
  },
  desktop: {
    breakpoint: { max: 1200, min: 1025 },
    items: 8,
    partialVisibilityGutter: 0
  },
  laptop: {
    breakpoint: { max: 1024, min: 769 },
    items: 8,
    partialVisibilityGutter: 0
  },  
  tablet: {
    breakpoint: { max: 768, min: 481 },
    items: 6,
    partialVisibilityGutter: 0
  },
  mobile: {
    breakpoint: { max: 480, min: 0 },
    items: 4,
    partialVisibilityGutter: 0
    }

    
};


const Tuner = ({onStationLogoClick}) => {

  const [countryCode, setCountryCode] = useState("GB")

  const [searchByNameFilter, setSearchByNameFilter] = useState("")
  const [stationFilter, setStationFilter] = useState({
    countryCode: `${countryCode}`,
    limit: 50,
    offset: 0,
    lastCheckOk: true
  });
  const [tunerDisplayData, setTunerDisplayData] = useState([])

  const SetDefaultSrc = (event) => {
		event.target.src = default_station_logo
	};

  const handleSearchByNameInput = (event) => {
    setSearchByNameFilter(event.target.value);
  };

  const handleSearchByNameSubmit = (event) => {
    event.preventDefault();
    if (setSearchByNameFilter === "") {
      return
    } else {
      setStationFilter({
      countryCode: `${countryCode}`,
      limit: 50,
      name: `${searchByNameFilter}`.toLowerCase(),
      offset: 0,
      lastCheckOk: true
    })
    setSearchByNameFilter("")
    }
  };

  const api = new RadioBrowserApi("BG Radio App")
//api.setBaseUrl('https://de1.api.radio-browser.info/')
//api.setBaseUrl('https://at1.api.radio-browser.info')
  api.setBaseUrl('https://nl1.api.radio-browser.info')

  const setupAPI = useCallback(async (stationFilter) => {
    return api.searchStations(stationFilter)
  },[])


  useEffect(() => {
    setupAPI(stationFilter)
    .then((data) => setTunerDisplayData(data))
    .catch(error => {console.log(error)})
  }, [setupAPI, stationFilter])

  const ButtonGroup = ({ next, previous}) => {
    return (
      <div className="tuner__ButtonGroup">
        <IconContext.Provider value={{ color: "whitesmoke", size: "36px", padding: "8px"}}>
          <BsArrowLeftSquareFill className="tuner__button" onClick={() => previous()}/>
          <BsArrowRightSquareFill className="tuner__button" onClick={() => next()}/>
        </IconContext.Provider>
      </div>
    )
  }




  return (
    <div className='tuner__container'>
      <div className='tuner__title'>
        <div>
          RADIOGRAM STEREO TUNER - 150620
        </div>
        <div>
        <form className='tuner__search-form' onSubmit={handleSearchByNameSubmit}>
          <input
            className='search-form__input'
            type='text'
            placeholder='Search station names...'
            value={searchByNameFilter}
            onChange={handleSearchByNameInput}
          />
          <button className='search-form__button' type='submit'>Go</button>  
        </form>
      </div>
        
      </div>
      <div className='tuner__main'>


{/* ### FILTERS ###   */}

      <div className="tuner-filter__container">
        {radio_search_filters.map((filter) => (
          <div
            key={filter}
            id={filter}
            className="tuner-filter__item"
              
            onClick={() => setStationFilter({
              countryCode: `${countryCode}`,
              limit: 50,
              tag: `${filter}`.toLowerCase(),
              offset: 0,
              lastCheckOk: true
              })                
            }
          >
            <button type="button" className="filter__button">
              <div className="filter__button-top">{filter.toUpperCase()}</div>
              <div className="filter__button-bottom"></div>
              <div className="filter__button-base"></div>
            </button> 
          </div>
          ))}
      </div>

      <div className='tuner-carousel'>
        <Carousel
          responsive={responsive}
          infinite={true}
          slidesToSlide = {8}
          renderButtonGroupOutside={true}
          customButtonGroup={<ButtonGroup/>}
          arrows={false}
          showDots
          >

        {tunerDisplayData.map((station) => (
          <div className='carousel-entry' key={station.id}>
            <img
              id={station.id}
              className="tuner-station__logo"
              name={station.name}
              src={station.favicon}
              data-urlresolved={station.urlResolved}
              alt={station.name}
              draggable="false"
              onClick ={onStationLogoClick}           
              
              onError={SetDefaultSrc}            
            />
            <div className='tuner-station__name'>
              {station.name.length > 28
              ?
              `${station.name.substring(0,28)}...`
              :
              station.name}
            </div>
          </div>
        ))}
        </Carousel>
      </div>
    </div>
    
    </div>
  )
}

export default Tuner