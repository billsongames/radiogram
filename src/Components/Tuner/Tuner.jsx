import { React, useEffect, useState } from 'react';
import { RadioBrowserApi } from "radio-browser-api";
import { Carousel } from '@trendyol-js/react-carousel';
import { api_test_data } from "../../data/api_test_data";

import "./tuner.css"

const api = new RadioBrowserApi('RadioPlayer')

const response = await api.searchStations({
  countryCode: 'US',
  limit: 10,
  offset: 0 // this is the default - can be omited
})

function Tuner() {

  const [tunerDisplayData, setTunerDisplayData] = useState([])

  const no_image = "./no_image_available.png"



  for (let i=0; i<response.length; i++) {
    let stationLogo
    let stationName

    if (response[i].favicon === "") {
      stationLogo = no_image
    } else {
      stationLogo = (response[i].favicon)
    }
    stationName = (response[i].name)

    tunerDisplayData.push({favicon: stationLogo, name: stationName})
  }  


    



  




  return(
    <div className='tuner'>
      <div className='tuner-carousel'>
        <Carousel
          show = {6}
          slide = {6}
          >

          {tunerDisplayData.map((station) => (
            <div className='carousel-entry'key = {station.name}>
              <div>
                <img className="station-logo" src={station.favicon} alt={station.name} />
              </div>
              <div>
                {station.name}
              </div>
            </div>
          ))}
        </Carousel>

      </div>
    </div>
    
  )
}  


export default Tuner

