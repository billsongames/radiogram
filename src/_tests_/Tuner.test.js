import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";

import Tuner from "../Components/Radio/Tuner/Tuner";

describe("Tuner", () => {
  const validProps= {
    tunerDisplayData: [
      {
        id: "test id",
        name: "test name 1",
        favicon: "test_favicon 1.jpg",
        urlResolved: "test url 1"
      }
    ],
    onStationLogoClick: jest.fn()
  }


  test("renders correctly" ,() => {
    const { asFragment } = render(<Tuner
      onStationLogoClick={validProps.onStationLogoClick}
    />)

    expect(asFragment()).toMatchSnapshot()
  })


  xtest("tuner renders correctly", () => {

  })

  test("filters render correctly", () => {

  })

  test("handles search submit correctly", () => {

  })

  xtest("click logo calls correct function", () => {

  })
})  