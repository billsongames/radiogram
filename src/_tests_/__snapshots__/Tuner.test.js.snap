// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tuner renders correctly 1`] = `
<DocumentFragment>
  <div
    class="tuner__container"
  >
    <div
      class="tuner__title"
    >
      <div>
        STEREO TUNER
      </div>
      <div>
        <form
          class="tuner__search-form"
        >
          <input
            class="search-form__input"
            placeholder="Search station names..."
            type="text"
            value=""
          />
          <button
            class="search-form__button"
            type="submit"
          >
            Go
          </button>
        </form>
      </div>
    </div>
    <div
      class="tuner__main"
    >
      <div
        class="tuner-filter__container"
      >
        <div
          class="tuner-filter__item"
          id="Pop"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              POP
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Rock"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              ROCK
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Metal"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              METAL
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Dance"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              DANCE
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Jazz"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              JAZZ
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Electro"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              ELECTRO
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Blues"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              BLUES
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Funk"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              FUNK
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Alternative"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              ALTERNATIVE
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
        <div
          class="tuner-filter__item"
          id="Classical"
        >
          <button
            class="filter__button"
            type="button"
          >
            <div
              class="filter__button-top"
            >
              CLASSICAL
            </div>
            <div
              class="filter__button-bottom"
            />
            <div
              class="filter__button-base"
            />
          </button>
        </div>
      </div>
      <div
        class="tuner-carousel"
      >
        <div
          class="react-multi-carousel-list  "
          dir="ltr"
        >
          <ul
            class="react-multi-carousel-track "
            style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
